### PR TITLE
Jhbate/upload meta check

### PR DESCRIPTION
**Pull request by jh-bate**
WIP but should be done, discussing with @jebeck

### CODE REVIEW

File: lib/core/api.js [hunk 1]
```diff
@@ -217,23 +212,13 @@ function postBlockToPlatform(data, groupId, callback) {
...
-    api.log('postBlockToPlatform: as logged in user');
-    tidepool.uploadDeviceDataForUser(data, function(err, result) {
-      if (err) {
-        return sad(err, err);
-      }
-      return happy(result);
-    });
-  }
+
+  tidepool.uploadDeviceDataForUser(data, groupId, function(err, result) {
```
Comment (jebeck) [inline lib/core/api.js]: Do you want to put an api.log back here? Maybe including logging of the groupId?

**Comment by jebeck** [comment]
This did break the CLIs @jh-bate 

Can you fix please?

**Comment by jh-bate** [comment]
sorry @jebeck - after saying I would test that I totally forgot about it.

**Comment by jh-bate** [comment]
So both cli's now pass in the id of the user into the driver.

**Comment by jebeck** [comment]
One of the log statements ends up with `undefined` both in CLIs and through UI: Api              | saved upload metadata  undefined

Trying to log the count?

**Comment by jh-bate** [comment]
yeah it was but  we just care if there is an error so updated that

**Comment by jebeck** [comment]
So this mostly works as I'd expect now. There's still an undefined being logged - maybe it's just a common issue for all the drivers? It occurs [here](https://github.com/tidepool-org/chrome-uploader/blob/jhbate/upload-meta-check/lib/drivers/carelinkDriver.js#L314) and [here](https://github.com/tidepool-org/chrome-uploader/blob/jhbate/upload-meta-check/lib/drivers/insuletDriver.js#L1632), for instance...

**Comment by jebeck** [comment]
Oh and also the UTC time isn't being added to these errors...is that expected?

I think it'd be great if we could customize the error message for these too. Is that possible? "Upload failed due to bad metadata" or something?

**Comment by jh-bate** [comment]
Yeah I think there are a few places like that in the uploader as there was a lot of logging done in order to see what was going on.

With regards to the other Q. Yes, I was going to tackle that as part of the card https://trello.com/c/FccsTSO7. If your happy with that answer then shall we merge this and tackle the nicer consistent messages for the next card

**Comment by jebeck** [comment]
Leaving nicer messages for the next task is fine, but maybe remove the attempts at logging `result` in the drivers? I'd rather that not get left behind - I find it confusing (and potentially red herring) when there are undefined things being logged.

**Comment by jh-bate** [comment]
cool - log statement removed from all drivers when there is an error

**Pull request merged**